dbms/src/Common/MyDuration.cpp

```diff
@@ -67,4 +67,10 @@ String MyDuration::toString() const
     auto frac_str = fmt::format("{:06}", microsecond);
     return fmt::format(fmt_str, sign > 0 ? "" : "-", hour, minute, second, frac_str);
 }
-} // namespace DB
\ No newline at end of file
+
+UInt64 calcSeconds(int hour, int minute, int second)
+{
+    return hour * 3600 + minute * 60 + second;
```

Comment:
There could be intermediate overflows during calculating `hour * 3600` or `minute * 60`. We should static cast them to `UInt64` first.